ui/components/app/wallet-overview/coin-overview.tsx

```diff
@@ -0,0 +1,472 @@
+  ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
+  const location = useLocation();
+  const metaMetricsId = useSelector(getMetaMetricsId);
+  const keyring = useSelector(getCurrentKeyring);
+  const usingHardwareWallet = isHardwareKeyring(keyring?.type);
+  ///: END:ONLY_INCLUDE_IF
+  const showFiat = useSelector(getShouldShowFiat);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+  const { ticker, type, rpcUrl } = useSelector(getProviderConfig);
+  const isOriginalNativeSymbol = useIsOriginalNativeTokenSymbol(
+    chainId,
+    ticker,
+    type,
+    rpcUrl,
+  );
+
+  const isExternalServicesEnabled = useSelector(getUseExternalServices);
+
+  const buttonTooltips: Record<
+    string,
+    { condition: boolean; message: string }[]
+  > = {
+    buyButton: [
+      ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
+      { condition: !isBuyableChain, message: '' },
+      ///: END:ONLY_INCLUDE_IF
+      { condition: !isSigningEnabled, message: 'methodNotSupported' },
+    ],
+    sendButton: [
+      { condition: !isSigningEnabled, message: 'methodNotSupported' },
+    ],
+    swapButton: [
+      { condition: !isSwapsChain, message: 'currentlyUnavailable' },
+      { condition: !isSigningEnabled, message: 'methodNotSupported' },
+    ],
+    bridgeButton: [
+      ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
+      { condition: !isBridgeChain, message: 'currentlyUnavailable' },
+      ///: END:ONLY_INCLUDE_IF
+      { condition: !isSigningEnabled, message: 'methodNotSupported' },
+    ],
+  };
+
+  const generateTooltip = (buttonKey: string, contents: React.ReactElement) => {
+    const conditions = buttonTooltips[buttonKey];
+    const tooltipInfo = conditions.find(({ condition }) => condition);
+    if (tooltipInfo?.message) {
+      return (
+        <Tooltip title={t(tooltipInfo.message)} position="bottom">
+          {contents}
+        </Tooltip>
+      );
+    }
+    return contents;
+  };
+
+  ///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
+  const mmiPortfolioEnabled = useSelector(getMmiPortfolioEnabled);
+  const mmiPortfolioUrl = useSelector(getMmiPortfolioUrl);
+
+  const portfolioEvent = () => {
+    trackEvent({
+      category: MetaMetricsEventCategory.Navigation,
+      event: MetaMetricsEventName.MMIPortfolioButtonClicked,
+    });
+  };
+
+  const stakingEvent = () => {
+    trackEvent({
+      category: MetaMetricsEventCategory.Navigation,
+      event: MetaMetricsEventName.MMIPortfolioButtonClicked,
+    });
+  };
+
+  const renderInstitutionalButtons = () => {
+    return (
+      <>
+        <IconButton
+          className={`${classPrefix}-overview__button`}
+          Icon={<Icon name={IconName.Stake} color={IconColor.primaryInverse} />}
+          label={t('stake')}
+          onClick={() => {
+            stakingEvent();
+            global.platform.openTab({
+              url: `${mmiPortfolioUrl}/stake`,
+            });
+          }}
```

Comment:
We should avoid passing inline functions and use the `useCallback` hook

Reply:
Sounds like a good plan, lemme do all of this for every buttons!